modules/javafx.controls/src/main/java/javafx/scene/control/TextInputControl.java

```diff
@@ -181,6 +181,9 @@ private void updateSelectedText() {
             } else {
                 int start = sel.getStart();
                 int end = sel.getEnd();
+                int length = txt.length();
+                if (end > start + length) end = length;
```

Comment:
would it be possible to surround both conditional statements with { }'s?

Reply:
done